herbert_core/utilities/classes/@fast_map/private/optimize_.m

```diff
@@ -0,0 +1,15 @@
+function obj = optimize_(obj)
+%OPTIMIZE_  % place values into expanded array or cellarray, containing
+% NaN or empty where keys are missing and values where
+% keys are present. This array/cellarray is optimal for fast
+% access to the values as function of keys.
```

Comment:
what is an expanded array/cellarray - what does it contain? what are we doing putting NaNs in it in places? Why is it optimal? Why do you assume that your code is sufficiently transparent to be self-explaining? Please document it.

Reply:
optimal, because I have measured performance and found that this solution is the best. Here I state this fact, and this is how code description works. 

there is array, containing values where the keys are and there are nan-s where they are not. 

The description is already the size of the code itself -- I have described intentions and how it is done -- is in the code. I just do not understand what else I can write to make it clearer.